packages/temporal-client/src/types.ts

```diff
@@ -58,18 +73,45 @@ export const WorkflowDurationOptions = S.struct({
    *
    * @format number of milliseconds or {@link https://www.npmjs.com/package/ms | ms-formatted string}
    */
-  workflowExecutionTimeout: S.optional(S.union(S.string, S.number)),
+  workflowExecutionTimeout: S.optional(S.union(MsStr, S.number)),
 
   /**
    * Maximum execution time of a single workflow task. Default is 10 seconds.
    *
    * @format number of milliseconds or {@link https://www.npmjs.com/package/ms | ms-formatted string}
    */
-  workflowTaskTimeout: S.optional(S.union(S.string, S.number)),
+  workflowTaskTimeout: S.optional(S.union(MsStr, S.number)),
 })
 
 export const BaseWorkflowOptions = S.struct({
   workflowIdReusePolicy: S.optional(S.enums(WorkflowIdReusePolicy)),
+  /**
+   * Optional cron schedule for Workflow. If a cron schedule is specified, the Workflow will run as a cron based on the
+   * schedule. The scheduling will be based on UTC time. The schedule for the next run only happens after the current
+   * run is completed/failed/timeout. If a RetryPolicy is also supplied, and the Workflow failed or timed out, the
+   * Workflow will be retried based on the retry policy. While the Workflow is retrying, it won't schedule its next run.
+   * If the next schedule is due while the Workflow is running (or retrying), then it will skip that schedule. Cron
+   * Workflow will not stop until it is terminated or cancelled (by returning temporal.CanceledError).
+   * https://crontab.guru/ is useful for testing your cron expressions.
+   */
+  chronSchedule: S.optional(S.string),
+
+  /**
+   * Specifies additional non-indexed information to attach to the Workflow Execution. The values can be anything that
+   * is serializable by {@link DataConverter}.
+   */
+
+  memo: S.optional(S.record(S.string, S.unknown)),
+  /**
+   * Specifies additional indexed information to attach to the Workflow Execution. More info:
+   * https://docs.temporal.io/docs/typescript/search-attributes
+   *
+   * Values are always converted using {@link JsonPayloadConverter}, even when a custom data converter is provided.
+   */
+  searchAttributes: S.optional(SearchAttributes),
```

Comment:
The code I introduced to set `OrgId` uses a ternary to conditionally assign `searchAttributes` to `undefined`. Is S.optional going to cause decode/parse errors if it encounters `undefined`?

Reply:
Nope. We don't use exact optional property types or S.optional(S.string,{exact:true}) in this repo. So the production is this:

```typescript
readonly searchAttributes?: {
        readonly [x: string]: readonly number[] | readonly string[] | readonly boolean[];
    } | undefined;
```

Instead of this:
```typescript
readonly searchAttributes?: {
        readonly [x: string]: readonly number[] | readonly string[] | readonly boolean[];
    };
```

To produce # 2 you need both enabled